moveit_planners/ompl/ompl_interface/test/test_state_validity_checker.cpp

```diff
@@ -0,0 +1,361 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020, KU Leuven
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Jeroen De Maeyer */
+/**
+ *    This test checks the basics of a StateValidityChecker:
+ *        - Can we create one?
+ *        - States inside and outside joint limits.
+ *        - States that are in self-collision.
+ *
+ *    It does not yet test:
+ *        - Collision with objects in the environment.
+ *        - Any type of path constraints.
+ *
+ *    The test do show what is minimally required to create a working StateValidityChecker.
+ **/
+
+#include "load_test_robot.h"
+
+#include <limits>
+#include <ostream>
+
+#include <gtest/gtest.h>
+
+#include <moveit/ompl_interface/detail/state_validity_checker.h>
+#include <moveit/ompl_interface/model_based_planning_context.h>
+#include <moveit/ompl_interface/parameterization/joint_space/joint_model_state_space.h>
+#include <moveit/planning_scene/planning_scene.h>
+
+#include <ompl/geometric/SimpleSetup.h>
+
+/** \brief Use this flag to turn on extra output on std::cout for debugging. **/
+constexpr bool VERBOSE{ false };
```

Comment:
Why not use ROS logging macros and setting them to output DEBUG with a namespace for this test?

Reply:
True! It's an old habit from when I didn't bother setting up my ROS debug levels correctly... I will fix this.

Reply:
Hmm, I will change the log statements, but I also use it to set the verbosity of the state validity checker
```C++
checker->setVerbose(VERBOSE);
```
which in turn is passed on to the collision request. Maybe I can make it an optional command-line argument for the test. The verbose output from the collision checker can be really useful when debugging. I can also set it to true by default.

Reply:
Talking about "minor" patches, it's really annoying that we can't specify arbitrary output streams for these methods. :-)